python/lsst/ctrl/mpexec/mpGraphExecutor.py

```diff
@@ -168,11 +168,19 @@ def _executeJob(
 
         quantum = pickle.loads(quantum_pickle)
         try:
-            quantumExecutor.execute(task_node, quantum)
+            _, report = quantumExecutor.execute(task_node, quantum)
+        except Exception as exc:
+            _LOG.debug("exception from task %s dataId %s: %s", task_node.label, quantum.dataId, exc)
+            report = QuantumReport.from_exception(
+                exception=exc,
+                dataId=quantum.dataId,
+                taskLabel=task_node.label,
+            )
+            raise
         finally:
             # If sending fails we do not want this new exception to be exposed.
             try:
-                report = quantumExecutor.getReport()
+                _LOG.debug("sending report for task %s dataId %s", task_node.label, quantum.dataId)
                 snd_conn.send(report)
             except Exception:
                 pass
```

Comment:
Wait... you mean you're _relying_ on this block to catch the `UnboundLocalError` if `report` is not defined? 😬

Reply:
Sorry, my previous reply went to a different comment, this is now fixed on latest commit.